### PR TITLE
epa-ng + raxml-ng makefiles: created targets and added LGG4F model target

### PR DESCRIPTION
**Pull request by MichelleHagen**

### CODE REVIEW

File: lib/make/makefile.epa-ng [hunk 1]
```diff
@@ -13,22 +13,36 @@ SHELL := /bin/bash
...
 
 # *** Targets ***
 
-%.suffix0: %.suffix1 %.suffix1 %.suffixn
-	@$(MAKECALL___PGM__)
-	__PGM_CALL__
-	@echo "`date +"%Y%m%d %H:%M:%S"`: DONE" >> $@.makecall
+# Target to rename the Reference Alignment, if it has been reduced by RAxML to match the epa_result target
+%.raxml.alnfaa: %.raxml.reduced.alnfaa
+        mv $(basename $(basename $@)).raxml.reduced.alnfaa $@
```
Comment (erikrikarddaniel) [inline lib/make/makefile.epa-ng]: I'm not sure about this one. First, there's a variable for the first dependency: `$<`, so you could replace the double `basename` called. Second, I think this is better to do manually (or have in your own `Makefile`) as the name of the actual input might differ; the user might not even have run RAxML-NG. If one would have this, it would be best to have also the `.phy` to `.alnfaa` target. All in all, I suggest just to remove the target and live with the inconvenience of having to do some things manually.

File: lib/make/makefile.epa-ng [hunk 2]
```diff
@@ -13,22 +13,36 @@ SHELL := /bin/bash
...
+# - ReferenceTree (rooted and safed as newick file)
+# - ReferenceAlignment (with the same id as the RAxML RT, raxml .phy file has been converted back to .alnfaa)
+# - QueryAlignment (created with hmmer)
+# 
+# epa-ng creates pre-named output files in the current directory:
+# - epa_info.log
+# - epa_results.jplace
+# so RT, RA and QS have to be in the same directory with no other RT, RA or QS files!
+
+epa_result.jplace: $(wildcard *.rooted.newick) $(wildcard *.raxml.alnfaa) $(wildcard *.hmmer.alnfaa)
```
Comment (erikrikarddaniel) [inline lib/make/makefile.epa-ng]: The `wildcard` calls here are expected to find just one file each. This is risky, as more than one file might match, and your numbering below is not correct. The problem you're trying to solve is, I believe, that the output file for the run is generic (`epa_result.jplace`). There are possibly two ways to solve this so you can turn it into a pattern target instead (i.e. with `%` in names). First, and easiest, there might be a flag with which you can specify the name of the output. If there isn't, you can rename the output file after a successful run, i.e. add a line `mv epa_result.jplace $@`. The disadvantage of the second approach is that it makes it impossible or unsafe to run multiple runs in the same directory.

File: lib/make/makefile.raxml-ng [hunk 3]
```diff
@@ -36,38 +36,41 @@ MAKECALL_RAXML_NG             = $(MAKECALL_RAXML_NG_VERSION); $(MAKECALL_RAXML_N
 
 # Nucleotide targets
 %.GTRG8F.raxml.rba: %.alnfna
-	raxml-ng --parse --msa $< --model GTR+G8+F --prefix $(basename $(basename $@))
+        raxml-ng --parse --msa $< --model GTR+G8+F --prefix $(basename $(basename $@))
```
Comment (erikrikarddaniel) [inline lib/make/makefile.raxml-ng]: From here, and all the way down I believe, targets belong in `makefile.raxml-ng`. Do any modifications you need in that, and make a separate PR.
That you have included this here indicates that you don't know that you can `include` more than one library makefile in your `Makefile`.

**Review by erikrikarddaniel — COMMENTED** [review]
A few comments to sort out. Impressive that you deciphered my template! :-)

File: lib/make/makefile.epa-ng (hunk 1 above)
Comment (MichelleHagen) [reply lib/make/makefile.epa-ng]: Okay yes I get that. I will remove this target. I was just thinking of a nice way to not have name conflicts because of the reduced files and also because there are 2 .alnfaa files as input.

File: lib/make/makefile.epa-ng (hunk 2 above)
Comment (MichelleHagen) [reply lib/make/makefile.epa-ng]: I can only specify an output dir (at least that's all I could find out) so the output will always be named `epa_result.jplace ` but I did not think about renaming the output itself (but I will also have to mv the logfile as it also has the same name every time). I will add that. Also: My query file was not named with the same prefix as the other files but that was more of a lazy mistake from me and I just have to adjust the filenames which is overall the better and smarter way.

File: lib/make/makefile.raxml-ng (hunk 3 above)
Comment (MichelleHagen) [reply lib/make/makefile.raxml-ng]: The changes are in the makefile.raxml-ng, I just pushed both files at once and I guess thats also why the pull request included it in one request. Here it also says that this part is from [lib/make/makefile.raxml-ng]

**Review by erikrikarddaniel — APPROVED** [review]
LGTM. Do the merge yourself! (I'll give you permissions first; just a moment.)
